pkg/registration/spoke/spokeagent.go

```diff
@@ -170,10 +170,52 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 	namespacedManagementKubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(
 		managementKubeClient, 10*time.Minute, informers.WithNamespace(o.agentOptions.ComponentNamespace))
 
+	// bootstrapKubeConfig is the file path of the bootstrap kubeconfig. If MultipleHubs feature is enabled, it is
+	// the selected bootstrap kubeconfig file path.
+	//
+	// hubValiedConfig is a function to check if the hub client config is valid.
+	var bootstrapKubeConfig string
+	var hubValidConfig func(ctx context.Context) (bool, error)
+	if features.SpokeMutableFeatureGate.Enabled(ocmfeature.MultipleHubs) {
+		// bootstrapkubeconfig
+		var bootstrapKubeConfigs []*rest.Config
+		for _, fp := range o.registrationOption.BootstrapKubeconfigs {
+			kc, err := clientcmd.BuildConfigFromFlags("", fp)
+			if err != nil {
+				return fmt.Errorf("unable to load bootstrap kubeconfig from file %q: %w", fp, err)
+			}
+			bootstrapKubeConfigs = append(bootstrapKubeConfigs, kc)
+		}
+
+		index, err := selectBootstrapKubeConfigs(ctx, o.agentOptions.SpokeClusterName, bootstrapKubeConfigs)
```

Comment:
if so, why just let select fun return a *rest.Config?

Reply:
Some legacy code is relaying on the file type of bootstrapkubeconfig:
```
server, proxyURL, caData, err := parseKubeconfig(bootstrapKubeConfig)
```

Can I add a todo for this? I will to refactor the core logic of registration to depending on *rest.Config.

Reply:
yes, that is ok. Let's add a TODO